backends/audiocd/sdl/sdl-audiocd.cpp

```diff
@@ -67,44 +73,69 @@ bool SdlAudioCDManager::openCD(int drive) {
 	return (_cdrom != NULL);
 }
 
-void SdlAudioCDManager::stopCD() {
+void SdlAudioCDManager::close() {
+	DefaultAudioCDManager::close();
+
+	if (_cdrom) {
+                SDL_CDStop(_cdrom);
+                SDL_CDClose(_cdrom);
+		_cdrom = 0;
+        }
+}
+
+void SdlAudioCDManager::stop() {
+	DefaultAudioCDManager::stop();
+
 	// Stop CD Audio in 1/10th of a second
 	_cdStopTime = SDL_GetTicks() + 100;
 	_cdNumLoops = 0;
 }
 
-void SdlAudioCDManager::playCD(int track, int num_loops, int start_frame, int duration) {
-	if (!num_loops && !start_frame)
-		return;
+bool SdlAudioCDManager::play(int track, int numLoops, int startFrame, int duration, bool onlyEmulate) {
+	// Prefer emulation
+	if (DefaultAudioCDManager::play(track, numLoops, startFrame, duration, onlyEmulate))
+		return true;
 
-	if (!_cdrom)
-		return;
+	// If we're set to only emulate, or have no CD, return here
+	if (onlyEmulate || !_cdrom)
+		return false;
 
+	if (!numLoops && !startFrame)
+		return false;
+
+	// FIXME: Explain this.
```

Comment:
Endy added the line years ago, but could not remember why when questioned. Perhaps you'd like to have a shot at explaining it.